src/dbnode/storage/database_test.go

```diff
@@ -1032,6 +1033,104 @@ func TestWideQuery(t *testing.T) {
 	assert.Equal(t, exSpans, spanStrs)
 }
 
+func TestReadMismatches(t *testing.T) {
+	ctrl := xtest.NewController(t)
+	defer ctrl.Finish()
+
+	d, mapCh, _ := defaultTestDatabase(t, ctrl, BootstrapNotStarted)
+	defer func() {
+		close(mapCh)
+	}()
+
+	commitlog := d.commitLog
+	d.commitLog = nil
+	ns := dbAddNewMockNamespace(ctrl, d, "testns")
+	nsOptions := namespace.NewOptions()
+	ns.EXPECT().OwnedShards().Return([]databaseShard{}).AnyTimes()
+	ns.EXPECT().Tick(gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+	ns.EXPECT().ShardBootstrapState().Return(ShardBootstrapStates{}).AnyTimes()
+	ns.EXPECT().Options().Return(nsOptions).AnyTimes()
```

Comment:
Please check if all of these methods are actually needed for this test case. Same in `TestWideQuery`.

Reply:
These are methods to set up the ns; they are necessary here

Reply:
But the tests pass without `OwnedShards`, `Tick` and `ShardBootstrapState`.